src/Service.Tests/CosmosTests/SamplerTests.cs

```diff
@@ -0,0 +1,213 @@
+        [DataRow("/name", null, null, 15, DisplayName = "PartitionBasedSampler: Retrieve default value of 5 records per partition if count limit is not set.")]
+        [DataRow(null, 1, 0, 9, DisplayName = "PartitionBasedSampler: Retrieve 1 record per partition, ignoring day-based filtering when partition key path is not specified.")]
+        public async Task TestPartitionBasedSampler(string partitionKeyPath, int? numberOfRecordsPerPartition, int? maxDaysPerPartition, int expectedResultCount)
+        {
+            Mock<PartitionBasedSampler> partitionBasedSampler
+                = new(_containerWithNamePk, partitionKeyPath, numberOfRecordsPerPartition, maxDaysPerPartition, _mockLogger.Object);
+            if (maxDaysPerPartition is not null)
+            {
+                partitionBasedSampler
+                    .Setup<long>(x => x.GetTimeStampThreshold())
+                    .Returns((long)(_sortedTimespansNamePk[0] - maxDaysPerPartition));
+            }
+
+            List<JsonDocument> result = await partitionBasedSampler.Object.GetSampleAsync();
+
+            Assert.AreEqual(expectedResultCount, result.Count);
+        }
+
+        /// <summary>
+        /// Verifies that the <c>PartitionBasedSampler</c> correctly retrieves partition key paths from containers with different partition key configurations.
+        /// </summary>
+        /// <remarks>
+        /// This test ensures that the <c>PartitionBasedSampler</c> accurately identifies and returns partition key paths from containers.
+        /// It includes cases where containers have single and multiple partition key paths to validate correct functionality.
+        /// </remarks>
+        [TestMethod]
+        public async Task TestGetPartitionInfoInPartitionBasedSampler()
+        {
+            Mock<PartitionBasedSampler> partitionBasedSampler = new(_containerWithNamePk, null, 1, 1, _mockLogger.Object);
+            List<string> result = await partitionBasedSampler.Object.GetPartitionKeyPaths();
+            Assert.AreEqual("name", result[0]);
+
+            partitionBasedSampler = new(_containerWithIdPk, null, 1, 1, _mockLogger.Object);
+            result = await partitionBasedSampler.Object.GetPartitionKeyPaths();
+            Assert.AreEqual("id", result[0]);
+
+            // Check if partition key path is getting fetched correctly if customer has multiple partitions
+            Container _containerWithHPk = await _database.CreateContainerIfNotExistsAsync("newcontainerwithhpk", "/anotherPojo/anotherProp");
+            partitionBasedSampler = new(_containerWithHPk, null, 1, 1, _mockLogger.Object);
+            result = await partitionBasedSampler.Object.GetPartitionKeyPaths();
+            Assert.AreEqual("anotherPojo", result[0]);
+            Assert.AreEqual("anotherProp", result[1]);
+
+            await _containerWithHPk.DeleteContainerAsync();
```

Comment:
seems like three different tests testing different things. (yes, same overall goal of testing partitionbasedsampler) HOwever, these are unique tests for the partition based sampler. I'm thinking of it like this: if one of the tests falis, what does that indicate is wrong with the sampler? Most likely something wrong that is different than what the other few tests are checking.